localstack/services/kinesisfirehose/resource_providers/aws_kinesisfirehose_deliverystream.py

```diff
@@ -0,0 +1,492 @@
+    CommonAttributes: Optional[list[HttpEndpointCommonAttribute]]
+    ContentEncoding: Optional[str]
+
+
+class HttpEndpointDestinationConfiguration(TypedDict):
+    EndpointConfiguration: Optional[HttpEndpointConfiguration]
+    S3Configuration: Optional[S3DestinationConfiguration]
+    BufferingHints: Optional[BufferingHints]
+    CloudWatchLoggingOptions: Optional[CloudWatchLoggingOptions]
+    ProcessingConfiguration: Optional[ProcessingConfiguration]
+    RequestConfiguration: Optional[HttpEndpointRequestConfiguration]
+    RetryOptions: Optional[RetryOptions]
+    RoleARN: Optional[str]
+    S3BackupMode: Optional[str]
+
+
+class Tag(TypedDict):
+    Key: Optional[str]
+    Value: Optional[str]
+
+
+REPEATED_INVOCATION = "repeated_invocation"
+
+
+class KinesisFirehoseDeliveryStreamProvider(
+    ResourceProvider[KinesisFirehoseDeliveryStreamProperties]
+):
+
+    TYPE = "AWS::KinesisFirehose::DeliveryStream"  # Autogenerated. Don't change
+    SCHEMA = util.get_schema_path(Path(__file__))  # Autogenerated. Don't change
+
+    def create(
+        self,
+        request: ResourceRequest[KinesisFirehoseDeliveryStreamProperties],
+    ) -> ProgressEvent[KinesisFirehoseDeliveryStreamProperties]:
+        """
+        Create a new resource.
+
+        Primary identifier fields:
+          - /properties/DeliveryStreamName
+
+
+
+        Create-only properties:
+          - /properties/DeliveryStreamName
+          - /properties/DeliveryStreamType
+          - /properties/ElasticsearchDestinationConfiguration/VpcConfiguration
+          - /properties/AmazonopensearchserviceDestinationConfiguration/VpcConfiguration
+          - /properties/AmazonOpenSearchServerlessDestinationConfiguration/VpcConfiguration
+          - /properties/KinesisStreamSourceConfiguration
+
+        Read-only properties:
+          - /properties/Arn
+
+        IAM permissions required:
+          - firehose:CreateDeliveryStream
+          - firehose:DescribeDeliveryStream
+          - iam:GetRole
+          - iam:PassRole
+          - kms:CreateGrant
+          - kms:DescribeKey
+
+        """
+        model = request.desired_state
+        firehose = request.aws_client_factory.firehose
+        parameters = [
+            "DeliveryStreamName",
+            "DeliveryStreamType",
+            "S3DestinationConfiguration",
+            "ElasticsearchDestinationConfiguration",
+            "AmazonopensearchserviceDestinationConfiguration",
+            "DeliveryStreamEncryptionConfigurationInput",
+            "ExtendedS3DestinationConfiguration",
+            "HttpEndpointDestinationConfiguration",
+            "KinesisStreamSourceConfiguration",
+            "RedshiftDestinationConfiguration",
+            "SplunkDestinationConfiguration",
+            "Tags",
+        ]
+        attrs = util.select_attributes(model, params=parameters)
+
+        if not request.custom_context.get(REPEATED_INVOCATION):
+            response = firehose.create_delivery_stream(**attrs)
```

Comment:
What if the delivery stream already exists? We should check that it does not first, or else raise an error (assuming that's what AWS does)

Reply:
I'm assuming boto3 should cast exception here, causing the error if delivery stream exists.
I can test with actual CF on AWS and see how it handles usage of non-unique name